modules/react/select/lib/hooks/useSelectInput.ts

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import {composeHooks, createElemPropsHook} from '@workday/canvas-kit-react/common';
+import {useComboboxInput, useScrollToIndexOnClick} from '@workday/canvas-kit-react/combobox';
+import {useSelectModel} from './useSelectModel';
+
+/**
+ * `useSelectInput` extends {@link useComboboxInput useComboboxInput}  and adds type ahead functionality and select specific [keyboard support](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/).
+ */
+export const useSelectInput = composeHooks(
+  useComboboxInput,
+  useScrollToIndexOnClick,
+  createElemPropsHook(useSelectModel)((model, ref) => {
+    const keySofar = React.useRef('');
+    const timer = React.useRef<ReturnType<typeof setTimeout>>();
+
+    // Reset after each keystroke to support type ahead
+    const startClearKeysSoFarTimer = () => {
+      if (timer.current) {
+        clearTimeout(timer.current);
+      }
+      timer.current = setTimeout(() => {
+        keySofar.current = '';
+      }, 500);
+    };
+
+    // Based on the the key typed, this will try and find an index in the array where it matches the string
+    // so if a user types `de` it will try and find an index in the array whose id matches `de` like denver
+    const getIndexByStartString = (
+      startIndex: number,
+      startString: string,
+      endIndex: number = model.state.items.length,
+      ignoreDisabled: boolean = true
+    ): number => {
+      for (let i = startIndex; i < endIndex; i++) {
+        const label = model.state.items[i].id.toLowerCase();
+
+        if (label.indexOf(startString.toLowerCase()) === 0) {
+          if (
+            !ignoreDisabled ||
+            (ignoreDisabled && !model.state.nonInteractiveIds.includes(model.state.items[i].id))
+          ) {
+            return i;
+          }
+        }
+      }
+      return -1;
```

Comment:
maybe to use ternary and set condition as variable:
```
const isNotDisabled = label.indexOf(startString.toLowerCase()) === 0 && (!ignoreDisabled || !model.state.nonInteractiveIds.includes(model.state.items[i].id))

return isNotDisabled ? i : -1;

```

you have nested `if`s, so you can combine them. on line 40 you don't need to check `ignoreDisabled` because you're checking this on line 39

Reply:
This one turned a little tricky. It didn't quite work as I expected